### ISSUE TITLE
CMake cannot find some packages if CMAKE_INSTALL_PREFIX is used

### ISSUE THREAD

**Issue by piotrdz** [mxe/mxe#904]
I came across this bug when trying to compile Colobot project with MXE (https://github.com/colobot/colobot).

By accident, I used a different CMake command than usual, adding `-DCMAKE_INSTALL_PREFIX=...` to command line. The result was a very weird bug, where for some packages CMake complained about missing files, while for other packages everything worked fine.

For example, with `find_package(OpenAL)`:
```
$ cmake -DCMAKE_TOOLCHAIN_FILE=/opt/mxe/usr/i686-w64-mingw32.static/share/cmake/mxe-conf.cmake -DCMAKE_INSTALL_PREFIX=/work/colobot-install-mxe ../                         
(...)
CMake Error at /usr/share/cmake-3.3/Modules/FindPackageHandleStandardArgs.cmake:148 (message):
  Could NOT find OpenAL (missing: OPENAL_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake-3.3/Modules/FindPackageHandleStandardArgs.cmake:388 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.3/Modules/FindOpenAL.cmake:101 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:3 (find_package)
```

While with `find_package(PhysFS)`:
```
$ cmake -DCMAKE_TOOLCHAIN_FILE=/opt/mxe/usr/i686-w64-mingw32.static/share/cmake/mxe-conf.cmake -DCMAKE_INSTALL_PREFIX=/work/colobot-install-mxe ../
(...)
-- Found PhysFS: /opt/mxe/usr/i686-w64-mingw32.static/lib/libphysfs.a  
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/test/build
```

I found that a workaround in case of OpenAL is to patch `FindOpenAL.cmake`, adding `/` as additional search path for `OPENAL_INCLUDE_DIR`. But this makes it even more confusing, as PhysFS uses the exact same search paths and somehow works fine.

I don't necessarily need this to work, as we normally don't use `CMAKE_INSTALL_PREFIX` with MXE, but I find it interesting as to how such a bug can occur. I have no idea how the install prefix could influence searching for packages but there you have it.

**Comment by tonytheodore** [comment]
> I don't necessarily need this to work, ... but I find it interesting as to how such a bug can occur.

Indeed, thanks for reporting this. Solid cmake support is one of our long-term goals and all bugs are valuable input. 

We have [some preliminary work](https://github.com/mxe/mxe/tree/master/src/cmake), but it's looking like we [have to provide our own cmake](https://github.com/mxe/mxe/issues/852#issuecomment-140430207) (as we do with pkg-conf and, by default, qmake).

Stay tuned.

**Comment by uwehermann** [comment]
I think we're being hit by the same issue in the sigrok project, see respective bug report.

http://sigrok.org/bugzilla/show_bug.cgi?id=663#c4

We do need/use cmake for the PulseView GUI, so this currently prevents a successful Windows build.

**Comment by uwehermann** [comment]
Comment from the bug reporter was "All good after MXE revert to a previous Git commit
```git reset --hard 4a4b5dc1f59783a242273bd196626988c500deea```".

I haven't tested this myself, though.

**Comment by tonytheodore** [comment]
@piotrdz 
> I have no idea how the install prefix could influence searching for packages but there you have it.

Apparently, [it's added to the search path](https://cmake.org/cmake/help/v3.0/variable/CMAKE_INSTALL_PREFIX.html):

> The installation prefix is also added to CMAKE_SYSTEM_PREFIX_PATH so that find_package, find_program, find_library, find_path, and find_file will search the prefix for other software.

but `CMAKE_SYSTEM_PREFIX_PATH` doesn't take `CMAKE_FIND_ROOT_PATH` into account and isn't intended to be set directly. ~~I'm guessing that setting `CMAKE_INSTALL_PREFIX` overrides the one set in the toolchain~~ that doesn't explain PhysFS or why it worked previously...

[Setting `CMAKE_PREFIX_PATH`](https://github.com/mxe/mxe/commit/ba7c553c66ee9e76792613feb00054eda48b8539) fixes it here - can you add it to your toolchain file?

@uwehermann I didn't try building PulseView but can you see if the same fixes it?

**Comment by uwehermann** [comment]
Hm, doesn't appear to make a difference for me, but I only did a quick test.

I did a "git pull" and "make mxe-conf".

Here's the new $HOME/mxe-git/usr/i686-w64-mingw32.static/share/cmake/mxe-conf.cmake:

```
set(CMAKE_SYSTEM_NAME Windows)
set(MSYS 1)
set(BUILD_SHARED_LIBS OFF)
set(LIBTYPE STATIC)
set(CMAKE_PREFIX_PATH /home/uwe/mxe-git/usr/i686-w64-mingw32.static)
set(CMAKE_FIND_ROOT_PATH /home/uwe/mxe-git/usr/i686-w64-mingw32.static)
set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
set(CMAKE_C_COMPILER /home/uwe/mxe-git/usr/bin/i686-w64-mingw32.static-gcc)
set(CMAKE_CXX_COMPILER /home/uwe/mxe-git/usr/bin/i686-w64-mingw32.static-g++)
set(CMAKE_Fortran_COMPILER /home/uwe/mxe-git/usr/bin/i686-w64-mingw32.static-gfortran)
set(CMAKE_RC_COMPILER /home/uwe/mxe-git/usr/bin/i686-w64-mingw32.static-windres)
set(CMAKE_MODULE_PATH "/home/uwe/mxe-git/src/cmake/modules" ${CMAKE_MODULE_PATH}) # For mxe FindPackage scripts
set(CMAKE_INSTALL_PREFIX /home/uwe/mxe-git/usr/i686-w64-mingw32.static CACHE PATH "Installation Prefix")
set(CMAKE_BUILD_TYPE Release CACHE STRING "Debug|Release|RelWithDebInfo|MinSizeRel")
set(CMAKE_CROSS_COMPILING ON) # Workaround for http://www.cmake.org/Bug/view.php?id=14075
set(CMAKE_RC_COMPILE_OBJECT "<CMAKE_RC_COMPILER> -O coff <FLAGS> <DEFINES> -o <OBJECT> <SOURCE>") # Workaround for buggy windres rules

file(GLOB mxe_cmake_files
    "/home/uwe/mxe-git/usr/i686-w64-mingw32.static/share/cmake/mxe-conf.d/*.cmake"
)
foreach(mxe_cmake_file ${mxe_cmake_files})
    include(${mxe_cmake_file})
endforeach()
```

Still seeing the issue we have, but not entirely sure it's the same as this one (#904). In our case cmake is not finding two C shared libs that have pkg-config files (whereas it did that before a certain revision just fine). 

Here's the full script we use to build PulseView, maybe we need a change there either way?
http://sigrok.org/gitweb/?p=sigrok-util.git;a=blob;f=cross-compile/mingw/sigrok-cross-mingw

Specifically, the cmake call we end up using is this:

```
 $ cmake -DCMAKE_TOOLCHAIN_FILE=/home/uwe/mxe-git/usr/i686-w64-mingw32.static/share/cmake/mxe-conf.cmake -DCMAKE_INSTALL_PREFIX:PATH=$HOME/sr_mingw -DDISABLE_WERROR=y -DENABLE_TESTS=y .
```

**Comment by uwehermann** [comment]
Nevermind, I think this issue is *not* the same that I'm seeing. I filed #1023 for this, including a potential fix.
